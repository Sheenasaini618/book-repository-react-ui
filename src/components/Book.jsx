import * as React from 'react';
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import CardActions from '@mui/material/CardActions';
import Avatar from '@mui/material/Avatar';
import Typography from '@mui/material/Typography';
import { red } from '@mui/material/colors';
import ManualEditBook from './ManualEditBook';
import DeleteBook from './DeleteBook';


export default function Book(props) {

  var book = props.details
  console.log(book)
  return (
    <Card sx={{ maxWidth: 450, maxHeight: 500 }} style={{ marginTop: "10px" }}>
      <CardHeader
        avatar={
          <Avatar sx={{ bgcolor: red[500] }} aria-label="recipe">
            {book.title.charAt(0).toUpperCase()}
          </Avatar>
        }
        title={book.title}
        subheader={book.authors[0]}
      />
      <CardMedia
        component="img"
        sx={{ width: 151 }}
        height="194"
        image={book.imageLinks.smallThumbnail}
        alt="Paella dish"
      />
      <CardContent>
        <Typography variant="h6" color="text.secondary" align='justify' style={{ fontWeight: 600 }}>
          Price: â‚¹{book.price}.00
        </Typography>
        <Typography variant="h6" color="text.secondary" align='justify' style={{ fontWeight: 600 }}>
          Copies Available: {book.quantity}
        </Typography>
      </CardContent>
      <CardActions disableSpacing>
        <ManualEditBook details={book} />
        <DeleteBook details={book.id} />
      </CardActions>
    </Card>
  );
}
