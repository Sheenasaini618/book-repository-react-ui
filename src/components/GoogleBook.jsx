import * as React from 'react';
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import CardActions from '@mui/material/CardActions';
import Avatar from '@mui/material/Avatar';
import Typography from '@mui/material/Typography';
import { red } from '@mui/material/colors';
import AddGoogleBook from './AddGoogleBook';


export default function GoogleBook(props) {

  var book = props.details
  
  console.log(book)
  return (
    <Card >
      <CardHeader 
        avatar={
          <Avatar sx={{ bgcolor: red[500] }} aria-label="recipe">
            {book.volumeInfo.title.charAt(0).toUpperCase()}
          </Avatar>
        }
        title= {book.volumeInfo.title}
        subheader= {book.volumeInfo.authors}
      />
       <CardMedia
        component="img"
        sx={{ width: 151 }}
        height="194"
        image= {book.volumeInfo.imageLinks.smallThumbnail}
        alt="Paella dish"
      /> 
      <CardContent>
        {/* <Typography variant="body2" color="text.secondary" align='justify' style={{height : 70}}>
        {book.volumeInfo.description}
        </Typography> */}
        <Typography variant="h6" color="text.secondary" align='justify' style={{fontWeight : 600}}>
        </Typography>
        <Typography variant="h6" color="text.secondary" align='justify' style={{fontWeight : 600}}>
        Copies Available: {book.volumeInfo.quantity}
        </Typography>
      </CardContent>
      <CardActions disableSpacing>
      <AddGoogleBook details={book} />
      </CardActions>
    </Card>
  );
}
