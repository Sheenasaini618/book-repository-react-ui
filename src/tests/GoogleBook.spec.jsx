import React from 'react'
import Book from '../components/Book'
import '@testing-library/jest-dom'
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow, render } from 'enzyme'
import { configure } from 'enzyme';
import GoogleBook from '../components/GoogleBook';


configure({ adapter: new Adapter() });
describe('<GoogleBook />', () => {

    let props, component, wrapper;

    beforeEach(() => {
        // charAt: jest.fn()
        // toUpperCase: jest.fn()

        props = {
            details: {
                "volumeInfo": {
                  "id": 1,
                  "title": "Introduction to Probability",
                  "imageLinks": {
                    "smallThumbnail": "http://books.google.com/books/content?id=2HxYV9lWlEcC&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                    "thumbnail": "http://books.google.com/books/content?id=2HxYV9lWlEcC&printsec=frontcover&img=1&zoom=1&source=gbs_api"
                  },
                  "authors": [
                    "John E. Freund"
                  ],
                  "description": "Featured topics include permutations and factorials, probabilities and odds, frequency interpretation, mathematical expectation, decision making, postulates of probability, rule of elimination, much more. Exercises with some solutions. Summary. 1973 edition.",
                  "price": 12566,
                  "quantity": 3
                }
              }
        }

        component = shallow(<GoogleBook {...props} />);
    })
    
    it('should render book ', () => {
        const element = component.debug()
        expect(element).toMatchSnapshot();
    });
})