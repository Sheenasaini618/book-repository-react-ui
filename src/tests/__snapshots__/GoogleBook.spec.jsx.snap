// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleBook /> should render book  1`] = `
"<ForwardRef(Card) className=\\"cardWhole\\">
  <ForwardRef(CardHeader) avatar={{...}} title=\\"Introduction to Probability\\" subheader={{...}} />
  <ForwardRef(CardMedia) component=\\"img\\" sx={{...}} height=\\"194\\" image=\\"http://books.google.com/books/content?id=2HxYV9lWlEcC&printsec=frontcover&img=1&zoom=5&source=gbs_api\\" alt=\\"Paella dish\\" />
  <ForwardRef(CardContent)>
    <ForwardRef(Typography) variant=\\"h6\\" color=\\"text.secondary\\" align=\\"justify\\" style={{...}} />
    <ForwardRef(Typography) variant=\\"h6\\" color=\\"text.secondary\\" align=\\"justify\\" style={{...}}>
      Copies Available: 
      3
    </ForwardRef(Typography)>
  </ForwardRef(CardContent)>
  <ForwardRef(CardActions) disableSpacing={true}>
    <AddGoogleBook details={{...}} />
  </ForwardRef(CardActions)>
</ForwardRef(Card)>"
`;
